src/Adapter/Product/Repository/ProductSupplierRepository.php

```diff
@@ -36,24 +37,41 @@
 use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\Exception\CannotUpdateProductSupplierException;
 use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\Exception\ProductSupplierNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ValueObject\ProductSupplierId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
 use ProductSupplier;
 
 /**
  * Methods for accessing ProductSupplier data source
  */
 class ProductSupplierRepository extends AbstractObjectModelRepository
 {
+    /**
+     * @var Connection
+     */
+    private $connection;
```

Comment:
```suggestion
    protected $connection;
```

Reply:
This is a nice idea, I think there is the same issue in other repositories so I'll make a general PR for all of them once this one is merged 😉